faker/providers/address/en_IN/__init__.py

```diff
@@ -391,10 +393,187 @@ class Provider(AddressProvider):
         "West Bengal",
     )
 
+    states_abbr = (
+        "AP",
+        "AR",
+        "AS",
+        "BR",
+        "CG",
+        "GA",
+        "GJ",
+        "HR",
+        "HP",
+        "JH",
+        "KA",
+        "KL",
+        "MP",
+        "MH",
+        "MN",
+        "ML",
+        "MZ",
+        "NL",
+        "OD",
+        "PB",
+        "RJ",
+        "SK",
+        "TN",
+        "TG",
+        "TR",
+        "UK",
+        "UP",
+        "WB",
+    )
+
+    union_territories = (
+        "Andaman and Nicobar Islands",
+        "Chandigarh",
+        "Dadra and Nagar Haveli, Dadra & Nagar Haveli",
+        "Daman and Diu",
+        "Delhi, National Capital Territory of Delhi",
+        "Jammu and Kashmir",
+        "Ladakh",
+        "Lakshadweep",
+        ("Pondicherry, Puducherry"),
+    )
+
+    union_territories_abbr = (
+        "AN",
+        "CH",
+        "DN",
+        "DD",
+        "DL",
+        "JK",
+        "LA",
+        "LD",
+        "PY",
+    )
+
+    # https://en.wikipedia.org/wiki/Postal_Index_Number
+
+    # FIXME: Some states such as `BR/JH` / `UK/UP` have similar PIN code ranges
+    # FIXME: as mentioned in above link.
+
+    state_pincode = {
+        "AP": (510_000, 539_999),
+        "AR": (790_000, 792_999),
+        "AS": (780_000, 789_999),
+        "BR": (800_000, 859_999),
+        "CG": (490_000, 499_999),
+        "GA": (403_000, 403_999),
+        "GJ": (360_000, 399_999),
+        "HR": (120_000, 139_999),
+        "HP": (170_000, 179_999),
+        "JH": (800_000, 859_999),
+        "KA": (560_000, 599_999),
+        "KL": [(670_000, 681_999), (683_000, 699_999)],
```

Comment:
can we make all items lists of tuples?

Reply:
If I make them list of tuples, I don't see how it would make a difference since it's meant to be never changed (unless a new state is created). I don't mind changing it to list of tuples as well. I'm just asking for your view

Reply:
I'd rather have all valueswith the same structure

Reply:
thanks for the feedback @fcurella 
updated as you suggested